src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -133,6 +133,56 @@ public static decimal Abs(decimal value)
             return decimal.Abs(value);
         }
 
+        [Intrinsic]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static double Abs(double value)
+        {
+            // Implementation based on https://git.musl-libc.org/cgit/musl/tree/src/math/fabs.c
+
+            const ulong signMask = 0x7FFFFFFFFFFFFFFF;
+
+            if (Sse2.IsSupported)
+            {
+                return Sse2.And(
+                    Vector128.CreateScalarUnsafe(value),
+                    Vector128.Create(signMask).AsDouble()).ToScalar();
+            }
+            else if (AdvSimd.IsSupported)
+            {
+                return AdvSimd.And(
+                    Vector128.CreateScalarUnsafe(value),
+                    Vector128.Create(signMask).AsDouble()).ToScalar();
+            }
+
+            ulong raw = BitConverter.DoubleToUInt64Bits(value);
+            return BitConverter.UInt64BitsToDouble(raw & signMask);
+        }
+
+        [Intrinsic]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static float Abs(float value)
+        {
+            // Implementation based on https://git.musl-libc.org/cgit/musl/tree/src/math/fabsf.c
+
+            const uint signMask = 0x7FFFFFFF;
+
+            if (Sse.IsSupported)
+            {
+                return Sse.And(
+                    Vector128.CreateScalarUnsafe(value),
+                    Vector128.Create(signMask).AsSingle()).ToScalar();
+            }
+            else if (AdvSimd.IsSupported)
+            {
+                return AdvSimd.And(
+                    Vector128.CreateScalarUnsafe(value),
+                    Vector128.Create(signMask).AsSingle()).ToScalar();
+            }
```

Comment:
so you either need to implement constant folding for Sse.And and AdvSimd.And in JIT or keep [Intrinsic] and remove this SIMD paths

Reply:
ok I've reverted back to 97b4006.